### PR TITLE
Media models

### PR DESCRIPTION
**Pull request by tydillon**
- Adds New and Replace media data models (still needs to be consolidated)
- A bit of a reorg, I separated clinicalsummary and media into two different modules for scalability
- Test updates
   - Added tests against the models coming from redox
   - Added tests to check the additionalProperties fn
   - Wrote tests for new and replace media
   - Added a big 'all test suite' that lets you run all of the tests at once (unsure if this is actually best practice or not, but just seemed logical...)

### CODE REVIEW

File: src/main/java/redox/clinicalsummary/common/Category.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-package redox.datamodel.common;
+package redox.clinicalsummary.common;
```
Comment (jtravan3) [inline src/main/java/redox/clinicalsummary/common/Category.java:2]: I like the new package but let's do `redox.datamodel.clinicalsummary.common` instead

File: src/main/java/redox/media/newmedia/Authenticator.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
+
+package redox.media.newmedia;
```
Comment (jtravan3) [inline src/main/java/redox/media/newmedia/Authenticator.java:2]: Same here `redox.datamodel.media.newmedia`

File: src/test/java/redox/clinicalsummary/patientpush/PatientPushTest_Myra.java [hunk 3]
```diff
@@ -185,4 +185,9 @@ public void testPatientPush_VitalSignsText() {
     public void testPatientPush_VitalSigns() {
         Assert.notNull(patientPush.getVitalSigns(), "VitalSigns object not successfully created");
     }
+
+    @Test
+    public void testPatientQuery_AdditionalProperties() {
+        Assert.isTrue(!patientPush.getAdditionalProperties().isEmpty(), "Additional Properties object was created when it should not have been");
```
Comment (jtravan3) [inline src/test/java/redox/clinicalsummary/patientpush/PatientPushTest_Myra.java:191]: Double negative here. Try
```java
Assert.isFalse(patientPush.getAdditionalProperties().isEmpty()
```

File: src/test/java/redox/clinicalsummary/patientpush/PatientPushTest_Sample.java [hunk 4]
```diff
@@ -1,14 +1,13 @@
-package redox.datamodel.patientpush;
+package redox.clinicalsummary.patientpush;
```
Comment (jtravan3) [inline src/test/java/redox/clinicalsummary/patientpush/PatientPushTest_Sample.java:1]: Same thing
`redox.datamodel.clinicalsummary.patientpush`

File: src/test/java/redox/clinicalsummary/patientquery/PatientQueryTest_Harley.java [hunk 5]
```diff
@@ -1,31 +1,26 @@
...
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.junit.Before;
-import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.springframework.test.context.junit4.SpringRunner;
 import org.springframework.util.Assert;
 import org.springframework.util.ResourceUtils;
-//import redox.datamodel.patientpush.Encounter;
-//import redox.datamodel.patientpush.PatientPush;
```
Comment (jtravan3) [inline src/test/java/redox/clinicalsummary/patientquery/PatientQueryTest_Harley.java:12]: If these aren't needed, remove the comments

File: src/test/java/redox/clinicalsummary/patientquery/PatientQueryTest_Sample.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.util.Assert;
+import org.springframework.util.ResourceUtils;
+//import redox.datamodel.patientpush.Encounter;
+//import redox.datamodel.patientpush.PatientPush;
```
Comment (jtravan3) [inline src/test/java/redox/clinicalsummary/patientquery/PatientQueryTest_Sample.java:11]: Same here

File: src/test/java/redox/clinicalsummary/patientqueryresponse/PatientQueryResponseTest_Sample.java [hunk 7]
```diff
@@ -1,4 +1,4 @@
-package redox.datamodel.patientqueryresponse;
+package redox.clinicalsummary.patientqueryresponse;
```
Comment (jtravan3) [inline src/test/java/redox/clinicalsummary/patientqueryresponse/PatientQueryResponseTest_Sample.java:1]: Same here
`redox.datamodel.clinicalsummary.patientqueryresponse`

**Review by jtravan3 — CHANGES_REQUESTED** [review]
Let me know if you have any questions

File: src/test/java/redox/clinicalsummary/patientquery/PatientQueryTest_Harley.java (hunk 5 above)
Comment (tydillon) [reply src/test/java/redox/clinicalsummary/patientquery/PatientQueryTest_Harley.java]: I don't think this test exists anymore, I think it moved to PatientQueryTest_Sample

**Review by jtravan3 — APPROVED** [review]
LGTM ✅

**Pull request merged**